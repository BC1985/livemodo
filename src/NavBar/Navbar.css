/* ----Global settings----- */
* {
  box-sizing: border-box;
}

.nav-wrapper {
  color: white;
  text-decoration: none;
  display: flex;
  align-items: center;
  padding: 0 1;
  height: 50px;
  height: 2.5em;
}

.nav-links {
  justify-content: space-evenly;
}

nav {
  display: flex;
  flex-direction: row;
  background-image: linear-gradient(to left, rgb(58, 69, 83), rgb(18, 18, 19));
  color: white;
  align-items: center;
  justify-items: center;
}

li {
  list-style-type: none;
}

.nav-links ul {
  display: flex;
  justify-content: center;
  padding-inline-start: 0px;
  margin-right: 10px;
  font-size: larger;
}

.nav-links a {
  margin-right: 15px;
  display: flex;
  flex-direction: row;
  justify-content: center;
  position: relative;
}

.nav-links a::after {
  border-bottom: 1px solid white;
  content: "";
  width: 0;
  height: 0;
  display: inline-block;
  position: absolute;
  bottom: 0;
  left: 0;
  transition: 0.2s ease-in;
}
.nav-links a:hover::after {
  width: 100%;
}
#logo {
  color: rgb(182, 182, 192);
  margin-left: 20px;
}

/* ----------- iPhone 5, 5S, 5C and 5SE ----------- */

/* Portrait */
@media only screen and (min-device-width: 320px) and (max-device-width: 568px) and (-webkit-min-device-pixel-ratio: 2) and (orientation: portrait) {
  .nav-links {
    display: none;
  }
  #logo {
    margin: 0 23%;
  }
}

/* Landscape */
@media only screen and (min-device-width: 320px) and (max-device-width: 568px) and (-webkit-min-device-pixel-ratio: 2) and (orientation: landscape) {
  .nav-wrapper > a {
    display: none;
  }
  .spacer {
    flex: 1;
  }
  .hamburger-icon {
    display: none;
  }
}
/* ----------- iPhone 6, 6S, 7 and 8 ----------- */

/* Portrait */
@media only screen and (min-device-width: 375px) and (max-device-width: 667px) and (-webkit-min-device-pixel-ratio: 2) and (orientation: portrait) {
  #logo {
    margin: 0 25%;
  }
}

/* Landscape */
@media only screen and (min-device-width: 375px) and (max-device-width: 667px) and (-webkit-min-device-pixel-ratio: 2) and (orientation: landscape) {
  .spacer {
    flex: 1;
  }
  .hamburger-icon {
    display: none;
  }
}

/* ----------- iPhone 6+, 7+ and 8+ ----------- */

/* Portrait */
@media only screen and (min-device-width: 414px) and (max-device-width: 736px) and (-webkit-min-device-pixel-ratio: 3) and (orientation: portrait) {
  #logo {
    margin: 0 30%;
  }
}

/* Landscape */
@media only screen and (min-device-width: 414px) and (max-device-width: 736px) and (-webkit-min-device-pixel-ratio: 3) and (orientation: landscape) {
  .spacer {
    flex: 1;
  }
  .hamburger-icon {
    display: none;
  }
}

/* ----------- iPad 3, 4 and Pro 9.7" ----------- */

/* Portrait */
@media only screen and (min-device-width: 768px) and (max-device-width: 1024px) and (orientation: portrait) and (-webkit-min-device-pixel-ratio: 2) {
  .nav-links a {
    font-size: 1.2em;
  }
  .hamburger-icon {
    display: none;
  }
  .spacer {
    flex: 1;
  }
}

/* Landscape */
@media only screen and (min-device-width: 768px) and (max-device-width: 1024px) and (orientation: landscape) and (-webkit-min-device-pixel-ratio: 2) {
  .nav-links a {
    font-size: 1.2em;
  }

  .spacer {
    flex: 1;
  }
  .hamburger-icon {
    display: none;
  }
}

/* -------Desktops and laptops ----------- */
@media only screen and (min-width: 1024px) {
  .hamburger-icon {
    display: none;
  }
  .spacer {
    flex: 1;
  }
  .nav-links a:focus {
    color: #f3d112;
  }
}

@media screen and (min-device-width: 1600px) {
  .nav-wrapper {
    height: 30px;
  }

  .nav-links li {
    font-size: 0.8em;
  }
  .nav-links a:focus {
    color: #f3d112;
  }

  .hamburger-icon {
    display: none;
  }
  .spacer {
    flex: 1;
  }
}
