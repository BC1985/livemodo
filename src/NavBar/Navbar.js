import React from "react";
import { NavLink, withRouter } from "react-router-dom";
import "./Navbar.css";
import { AppBar, Toolbar, Typography } from "@material-ui/core/";
import { makeStyles } from "@material-ui/core/styles";
import SideDrawer from "../SideDrawer/SideDrawer";

export const showRoutes = routes => {
  return routes.map(r => (
    <li key={r.name}>
      <NavLink to={r.to} onClick={r.onClick}>
        {r.name}
      </NavLink>
    </li>
  ));
};
const Navbar = ({ routes, authenticateRoutes, isLoggedIn }) => {
  const useStyles = makeStyles(theme => ({
    title: {
      flexGrow: 1,
    },
    links: {
      [theme.breakpoints.down("sm")]: {
        display: "none",
      }
    }
  }));
  const classes = useStyles();

  return (
    <AppBar position="static">
      <Toolbar>
        <SideDrawer
          isLoggedIn={isLoggedIn}
          routes={routes}
          authenticateRoutes={authenticateRoutes}
        />
        <Typography variant="h6" className={classes.title}>
          Livemodo
        </Typography>
        <div className="spacer" />
        <div className={`nav-links ${classes.links}`}>
          <ul>{showRoutes(isLoggedIn ? authenticateRoutes : routes)}</ul>
        </div>
      </Toolbar>
    </AppBar>
  );
};

export default withRouter(Navbar);
