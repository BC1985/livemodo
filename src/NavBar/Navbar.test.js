import React from "react";
import ReactDOM from "react-dom";
import { MemoryRouter } from "react-router-dom";
import NavBar from "../NavBar/Navbar";
const routes = [
  {
    to: "/",
    name: "Home"
  },
  {
    to: "/browse",
    name: "Browse Reviews"
  },
  {
    to: "/login",
    name: "Log in"
  }
];
it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <MemoryRouter>
      <NavBar routes={routes} />
    </MemoryRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});
