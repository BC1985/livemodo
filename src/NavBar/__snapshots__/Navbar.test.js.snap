// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UI as exptected 1`] = `
<div
  className="container"
>
  <nav
    className="nav-wrapper"
    role="navigation"
  >
    <div
      className="hamburger-icon"
    >
      <button
        id="toggle-button"
      >
        <div
          style={
            Object {
              "background": "white",
              "height": "3px",
              "marginBottom": "5px",
              "transform": "none",
              "transformOrigin": "top left",
              "transition": "all 0.3s ease",
              "width": "20px",
              "zIndex": 10,
            }
          }
        />
        <div
          style={
            Object {
              "background": "white",
              "height": "3px",
              "opacity": 1,
              "transform": "none",
              "transition": "all 0.3s ease",
              "width": "20px",
              "zIndex": 10,
            }
          }
        />
        <div
          style={
            Object {
              "background": "white",
              "height": "3px",
              "marginTop": "5px",
              "transform": "none",
              "transformOrigin": "top left",
              "transition": "all 0.3s ease",
              "width": "20px",
              "zIndex": 10,
            }
          }
        />
      </button>
    </div>
    <div
      id="logo"
    >
      LIVEMODO
    </div>
    <div
      className="spacer"
    />
    <div
      className="nav-links"
    >
      <ul>
        <li>
          <a
            aria-current="page"
            className="active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Home
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            href="/browse"
            onClick={[Function]}
          >
            Browse Reviews
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            href="/login"
            onClick={[Function]}
          >
            Log in
          </a>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
